### ISSUE TITLE
FP16 warnings

### ISSUE THREAD

**Issue by 8enmann** [cybertronai/transformer-xl#2]
Seeing a lot of prints like
```
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([3, 512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([3])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([50257])
```
and
```
OVERFLOW! Skipping step. Attempted loss scale: 2097152.0, reducing to 1048576.0
skipped iteration!
```
See `/ncluster/runs.new/ben-bpe2/info.log`
Repro:
`python launch.py --config=one_machine_fp16_small --name ben-bpe2`

**Comment by darius-lam** [comment]
The overflow warning is a feature from the dynamic loss scaler: when an overflow is detected (from the current loss scale being too large, which is to be expected as training goes on), it will automatically half the loss scale (to a minimum of 1).  I can add an argument that allows you to set the initial loss scale.  It defaults to 2^32. 

The prints about the FP16_Optimizer are merely logs the type of the tensors it receives as a debugging tool.  I will add an argument that allows you to turn it off; right now you can manually turn it off. 

tl;dr the warnings are harmless and are merely for logging purposes
